CMS-Enterprise/mint-app
# [MINT-3242] Linking Action Menu Add Milestone & Add Sub-Category to Respective Modals && pass down prepopulated categories

# MINT-3242

## Description

* Add `fireAfterOpening` prop to Modal component to allow custom function to fire upon modal opening
* Created `MTOModalContext` to allow for easier state management and prevent prop drilling for `MTOTable` and `MTOTableActions` 
* Updated defaultValues for `AddCustomCategoryForm` and `AddCustomMilestoneForm` to take in prepopulated categoryID and optional subcategoryID





## How to test this change

1.  Navigate to a pre-populated MTO Table
2. Play with the "..." action menu
3. Verify things are functioning as expected
    a. `Add model milestone` on parent category to pop up milestone modal with primary category pre-populated
    b. `Add sub-category` on parent category to pop up custom category modal and with things pre-populated
    c. `Add model milestone` on **subcategory** to pop up milestone modal with primary category **AND SUBCATEGORY** pre-populated

## PR Author Checklist


- [] I have updated tests or written new tests as appropriate in this PR.


## PR Reviewer Guidelines

- It's best to pull the branch locally and test it, rather than just looking at the code online!
- When approving a PR, provide a reason _why_ you're approving it
  - e.g. "Approving because I tested it locally and all functionality works as expected"
  - e.g. "Approving because the change is simple and matches the Figma design"
- Don't be afraid to leave comments or ask questions, especially if you don't understand why something was done! (This is often a great time to suggest code comments or documentation updates)
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
